### PR TITLE
feat(KEXP): add presence

### PR DESCRIPTION
**Pull request by alejsalaz**
## Description 

- Add presence for [KEXP](https://kexp.org/)
- The presence updates based on whether the radio is being listened to or not
- The presence image is taken from the cover of the currently playing song; if not available, it uses the station logo
- It includes a button to listen along simultaneously
- If there's no song playing, only the show name is shown
- Given support for the following parts of the page: [archive](https://kexp.org/archive/), [podcasts](https://kexp.org/podcasts/), [shows](https://kexp.org/shows/), [djs](https://kexp.org/djs/), [schedule](https://kexp.org/schedule/)
- Shows information about articles if the user is reading one, take these articles as examples:
  - [50 Years of Hip-Hop](https://kexp.org/podcasts/50-years-of-hip-hop/) under Podcasts
  - [90.TEEN](https://kexp.org/shows/90.-teen/) under Shows
  - [Abbie](https://kexp.org/djs/abbie/) under DJs

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/K/KEXP/presence.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			play: "general.playing",
+			profile: "general.readingAbout",
+			reading: "general.readingAnArticle",
+			watching: "general.watching",
+		}),
+		currentPagePath = window.location.pathname,
+		presenceData: PresenceData = {
+			startTimestamp: timeElapsed,
+			largeImageKey: "https://i.imgur.com/uzXz274.png",
+			buttons: [{ label: "Check KEXP", url: "https://kexp.org" }],
```
Comment (theusaf) [inline websites/K/KEXP/presence.ts:22]: Buttons that direct to main pages are [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).

File: websites/K/KEXP/presence.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	timeElapsed = Math.floor(Date.now() / 1000);
+async function updatePresence() {
+	const strings = await presence.getStrings({
+			pause: "general.paused",
+			play: "general.playing",
+			profile: "general.readingAbout",
+			reading: "general.readingAnArticle",
+			watching: "general.watching",
+		}),
+		currentPagePath = window.location.pathname,
```
Comment (theusaf) [inline websites/K/KEXP/presence.ts:18]: This repository prefers `document.location` over `window.location`
```suggestion
		currentPagePath = document.location.pathname,
```

File: websites/K/KEXP/presence.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		presenceData.state = document.querySelector("h3")?.textContent;
+	} else {
+		const coverImage = document
+			.querySelector(".Player-coverImage")
+			?.getAttribute("src");
+
+		presenceData.details = document.querySelector("a.Player-show")?.textContent;
+		presenceData.largeImageKey = coverImage?.startsWith("https://")
+			? coverImage
+			: "https://i.imgur.com/RBf6mDU.png";
```
Comment (theusaf) [inline websites/K/KEXP/presence.ts:51]: Use the square image. Alternatively, use an if statement to conditionally set the `largeImageKey`
```suggestion
			: "https://i.imgur.com/uzXz274.png";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

File: websites/K/KEXP/metadata.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	"description": {
+		"en": "Digging indie gems & underground bangers? KEXP Seattle's your radio refuge.  DJs handpick a global mixtape, from garage rock to dream pop, all spun with heart and Seattle swagger. Tune in and discover your next obsession! ✨",
+		"es": "¿Te cansa la misma música aburrida? KEXP Seattle es tu nuevo refugio sonoro. DJs expertos te preparan una mezcla global, desde garage rock hasta dream pop, con todo el corazón y la onda de Seattle. ¡Sintoniza y descubre tu próxima obsesión! ✨"
+	},
+	"url": [
+		"www.kexp.org",
+		"kexp.org"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/uzXz274.png",
```
Comment (DarkVillager) [inline websites/K/KEXP/metadata.json:17]: Logo's must be 512x512

File: websites/K/KEXP/presence.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+const presence = new Presence({ clientId: "1138970637440917504" }),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	defaultImageUrl = "https://i.imgur.com/uzXz274.png",
```
Comment (DarkVillager) [inline websites/K/KEXP/presence.ts:3]: Ditto

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**